src/execution/operator/csv_scanner/buffer_manager/csv_file_handle.cpp

```diff
@@ -1,17 +1,27 @@
 #include "duckdb/execution/operator/csv_scanner/csv_file_handle.hpp"
 #include "duckdb/common/exception/binder_exception.hpp"
 #include "duckdb/common/numeric_utils.hpp"
+#include "duckdb/common/compressed_file_system.hpp"
+#include "duckdb/common/string_util.hpp"
 
 namespace duckdb {
 
 CSVFileHandle::CSVFileHandle(FileSystem &fs, Allocator &allocator, unique_ptr<FileHandle> file_handle_p,
                              const string &path_p, FileCompressionType compression)
-    : file_handle(std::move(file_handle_p)), path(path_p) {
+    : compression_type(compression), file_handle(std::move(file_handle_p)), path(path_p) {
 	can_seek = file_handle->CanSeek();
 	on_disk_file = file_handle->OnDiskFile();
 	file_size = file_handle->GetFileSize();
 	is_pipe = file_handle->IsPipe();
-	uncompressed = compression == FileCompressionType::UNCOMPRESSED;
+	if (compression_type == FileCompressionType::AUTO_DETECT) {
```

Comment:
This seems to duplicate logic that is in `VirtualFileSystem::OpenFile` - could we perhaps make `GetCompression()` a virtual method on the `FIleHandle` that defaults to `FileCompressionType::UNCOMPRESSED`?